goroutine/goroutine_workflow.go

```diff
@@ -0,0 +1,62 @@
+package goroutine
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	"go.temporal.io/sdk/workflow"
+)
+
+/**
+* This sample workflow demonstrates how to use multiple Temporal gorotinues (instead of native goroutine) to process a
+* a sequence of activities in parallel.
+* In Temporal workflow, you should create goroutines using workflow.Go method.
+ */
+
+// SampleGoroutineWorkflow workflow definition
+func SampleGoroutineWorkflow(ctx workflow.Context, parallelism int) (results []string, err error) {
+	ao := workflow.ActivityOptions{
+		StartToCloseTimeout: 10 * time.Second,
+	}
+	ctx = workflow.WithActivityOptions(ctx, ao)
+
+	for i := 0; i < parallelism; i++ {
+		input1 := fmt.Sprint(i) // Should be outside lambda to be captured correctly
+		// Start a goroutine in a workflow safe way
+		workflow.Go(ctx, func(gCtx workflow.Context) {
+			// It is important to use the context passed to the goroutine function
```

Comment:
> It is important to use the context passed to the goroutine function

I usually shadow the outer variable by giving the parameter the same name (`ctx` in this case) to avoid mistakes. But unsure if doing so detracts from the lesson.